storage/src/mmr/bitmap.rs

```diff
@@ -428,20 +421,30 @@ impl<H: CHasher, const N: usize> Bitmap<H, N> {
         let leaf_pos = Self::leaf_pos(bit_offset);
         let chunk = self.get_chunk(bit_offset);
 
+        if leaf_pos == self.mmr.size() {
+            assert!(self.next_bit > 0);
+            // Proof is over a bit in the partial chunk. In this case only a single digest is
+            // required in the proof: the mmr's root.
+            return Ok((
+                Proof {
+                    size: self.bit_count(),
+                    digests: vec![self.mmr.root(hasher)],
+                },
+                *chunk,
+            ));
+        }
+
+        let mut proof = Proof::<H>::range_proof(&self.mmr, leaf_pos, leaf_pos).await?;
+        proof.size = self.bit_count();
         if self.next_bit == 0 {
-            let proof = Proof::<H>::range_proof(&self.mmr, leaf_pos, leaf_pos).await?;
+            // Bitmap is chunk aligned.
             return Ok((proof, *chunk));
         }
 
-        // We must account for the bits in the last chunk.
-        let mut mmr = self.mmr.clone_pruned();
-        mmr.add(hasher, self.last_chunk()).await?;
-
-        let storage = BitmapStorage {
-            mmr: &self.mmr,
-            last_chunk_mmr: &mmr,
-        };
-        let proof = Proof::<H>::range_proof(&storage, leaf_pos, leaf_pos).await?;
+        let last_chunk_digest = hasher.digest(self.last_chunk());
+        // Since the bitmap wasn't chunk aligned, we'll need to include the digest of the last chunk
```

Comment:
nit: suggest moving to above 444 instead.

Reply:
done